pkg/parse/events/publisher.go

```diff
@@ -154,27 +155,29 @@ func (s *RetrySyncPublisher) Start(ctx context.Context) reflect.Value {
 // called and an empty Result is returned.
 func (s *RetrySyncPublisher) Publish(subscriber Subscriber) Result {
 	if s.currentBackoff.Steps == 0 {
-		klog.Infof("Retry limit (%v) has been reached", s.retryLimit)
+		klog.Infof("Retry limit has been reached (%v)", s.retryLimit)
 		// Don't reset retryTimer if retry limit has been reached.
 		return Result{}
 	}
 
-	retryDuration := s.currentBackoff.Step()
+	s.nextDelay = s.currentBackoff.Step()
 	retries := s.retryLimit - s.currentBackoff.Steps
-	klog.Infof("a retry is triggered (retries: %v/%v)", retries, s.retryLimit)
+	klog.V(3).Infof("Sending retry event (step: %v/%v)", retries, s.retryLimit)
```

Comment:
Should we log this message at the default level?

If no retry attempt is needed, we would only publish the retry event once, so the log will only show up once.

Reply:
Yeah. This is one of the log changes I added to the first commit, before I added the second commit with the channels. With the channels triggering SyncEvents instead of RetryEvents, we don't actually need to 'mute' this.